src/Components/Components/src/Routing/Router.cs

```diff
@@ -29,6 +32,10 @@ public class Router : IComponent, IHandleAfterRender, IDisposable
         bool _navigationInterceptionEnabled;
         ILogger<Router> _logger;
 
+        private CancellationTokenSource _onNavigateCts;
+
+        private HashSet<Assembly> _assemblies = new HashSet<Assembly>();
```

Comment:
Rather than replacing this instance outright, would it work to make this `private readonly` and mutate the existing instance?

Just so it's easier to reason about - there's no risk that anything ends up holding a reference to an orphaned instance.